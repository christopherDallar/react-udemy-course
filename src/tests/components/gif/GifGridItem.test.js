import React from 'react'
// import { render } from '@testing-library/react';
import { shallow } from 'enzyme'
import { GifGridItem } from '../../../shared/components/gif/GifGridItem.component'

describe('Test on GiftGridItem Component', () => {
  const title = 'Un titulo'
  const url = 'https://localhost/'

  const wrapper = shallow(<GifGridItem title={title} url={url} />)

  test('should Match With Snapshot GiftGridItem Correctly', () => {
    expect(wrapper).toMatchSnapshot() // To Save the literal rendered component and compare with wrapper if you press 'u' update the snapshot
  })

  test(`should has the same ${title} in tag <p> `, () => {
    const p = wrapper.find('p')
    expect(p.text().trim()).toBe(title)
  })

  test('should has the same src and alt of <img>', () => {
    const img = wrapper.find('img')
    expect(img.prop('src')).toBe(url)
    expect(img.prop('alt')).toBe(title)
  })

  test('should has class in <div> the className  animate__bounce', () => {
    const div = wrapper.find('div')
    const className = 'animate__bounce'
    expect(div.hasClass(className)).toBe(true)
  })
})
