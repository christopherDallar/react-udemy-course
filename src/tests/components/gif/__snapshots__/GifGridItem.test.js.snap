// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GiftGridItem Component should Match With Snapshot GiftGridItem Correctly 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Un titulo"
    src="https://localhost/"
  />
  <p>
     
    Un titulo
     
  </p>
</div>
`;
